File: app/src/main/java/com/pleon/buyt/database/Purchase.java
package com.pleon.buyt.database;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity
public class Purchase {

    @PrimaryKey(autoGenerate = true)
    public long id;

    @ForeignKey(entity = Shop.class, parentColumns = "id", childColumns = "shopId")
    public long shopId;

    public String date;

    public Purchase(long shopId, String date) {
        this.shopId = shopId;
        this.date = date;
    }
}

File: app/src/main/kotlin/com.pleon.buyt/model/Purchase.kt
package com.pleon.buyt.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import java.util.*

@Entity
class Purchase(@field:ForeignKey(entity = Store::class, parentColumns = ["id"], childColumns = ["storeId"])
               var storeId: Long, var date: Date?) {

    @PrimaryKey(autoGenerate = true)
    var purchaseId: Long = 0
}
